modules/core/src/main/java/org/locationtech/jts/index/strtree/STRtree.java

```diff
@@ -263,7 +263,30 @@ protected Comparator getComparator() {
     BoundablePair bp = new BoundablePair(this.getRoot(), this.getRoot(), itemDist);
     return nearestNeighbour(bp);
   }
-  
+  /**
+   * Finds k items in this tree which are the top k nearest neighbors to the given {@link Object}, 
+   * using {@link ItemDistance} as the distance metric.
+   * A Branch-and-Bound tree traversal algorithm is used
+   * to provide an efficient search. This method implements the KNN algorithm described in the following paper:
+   * Roussopoulos, Nick, Stephen Kelley, and Frédéric Vincent. "Nearest neighbor queries." ACM sigmod record. Vol. 24. No. 2. ACM, 1995.
+   * <p>
+   * The query <tt>object</tt> does <b>not</b> have to be 
+   * contained in the tree, but it does 
+   * have to be compatible with the <tt>itemDist</tt> 
+   * distance metric. 
+   * 
+   * @param env the envelope of the query item
+   * @param item the item to find the nearest neighbour of
+   * @param itemDist a distance metric applicable to the items in this tree and the query item
+   * @param k the K nearest items in kNearestNeighbour
+   * @return the K nearest items in this tree
+   */
+  public Object[] kNearestNeighbour(Envelope env, Object item, ItemDistance itemDist,int k)
```

Comment:
It might be better to call this nearestNeighbour (using overloading), or else nearestNeighbourK (so that the methods are adjacent in the Javadoc)